server/api/order_product.js

```diff
@@ -4,7 +4,23 @@ const { OrderProduct } = require('../db/models');
 // Updating the order_product relationship; adding a product to an order
 router.post('/', (req, res, next) => {
   OrderProduct.create(req.body)
-    .then(products => res.status(201).send(products))
+    .then(lineitem => res.status(201).send(lineitem))
+    .catch(next);
+});
+
+router.put('/', (req, res, next) => {
+  return OrderProduct.findOne({
+    where: {
+      orderId: req.body.orderId,
+      productId: req.body.productId,
+    },
+  })
+    .then(lineitem => {
+      return lineitem.update({
+        quantity: req.body.quantity,
+      });
+    })
+    .then(lineitem => res.status(201).send(lineitem))
```

Comment:
So I did some security checking when trying to implement it yesterday but I mean we can do that later.

```
  const putOrdProd = async () => {
    const order = await Order.findById(req.body.orderId);
    const { userId } = order;

    if (req.user && req.user.id === userId) {
      const orderProduct = await OrderProduct.findOne({
        where: {
          orderId: req.body.orderId,
          productId: req.body.orderId,
        },
      });
      await orderProduct.update({ quantity: res.body.quantity });
      res.sendStatus(204);
    } else {
      res.sendStatus(403);
    }
  };

  putOrdProd()
    .then(null, next);
}
```